ports/esp32/machine_rtc.c

```diff
@@ -118,27 +130,31 @@ STATIC mp_obj_t machine_rtc_init(mp_obj_t self_in, mp_obj_t date) {
     mp_obj_t args[2] = {self_in, date};
     machine_rtc_datetime_helper(2, args);
 
+    #if MICROPY_HW_RTC_USER_MEM_MAX > 0
     if (rtc_user_mem_magic != MEM_MAGIC) {
         rtc_user_mem_magic = MEM_MAGIC;
         rtc_user_mem_len = 0;
     }
+    #endif
+
     return mp_const_none;
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(machine_rtc_init_obj, machine_rtc_init);
 
+
+#if MICROPY_HW_RTC_USER_MEM_MAX > 0
 STATIC mp_obj_t machine_rtc_memory(mp_uint_t n_args, const mp_obj_t *args) {
     if (n_args == 1) {
         // read RTC memory
-        uint32_t len = rtc_user_mem_len;
-        uint8_t rtcram[MEM_USER_MAXLEN];
-        memcpy( (char *) rtcram, (char *) rtc_user_mem_data, len);
-        return mp_obj_new_bytes(rtcram,  len);
+        uint8_t rtcram[MICROPY_HW_RTC_USER_MEM_MAX];
+        memcpy( (char *) rtcram, (char *) rtc_user_mem_data, rtc_user_mem_len);
+        return mp_obj_new_bytes(rtcram,  rtc_user_mem_len);
```

Comment:
I would have thought that caching `rtc_user_mem_len` into the local variable `len` would have resulted in shorter code because it would only access the RTC memory once.  But maybe not?

Reply:
Building on my system I get
Direct access to RTC memory: 1396112
Indirect access through `len`: 1396128

Direct access to RTC memory is smaller (but might run negligibly slower)